cobbler/utils/__init__.py

```diff
@@ -1337,3 +1338,30 @@ def is_str_float(value: str) -> bool:
     except ValueError:
         pass
     return False
+
+
+@contextlib.contextmanager
+def filelock(lock_file: str):
+    """
+    Context manager to acquire a file lock and release it afterwards
+
+    :param lock_file: Path to the file lock to acquire
+    :raises CX: Raised in case of unexpect error acquiring file lock.
+    """
+    fd = os.open(lock_file, os.O_RDWR | os.O_CREAT | os.O_TRUNC, 0o660)
+    lock_file_fd = None
+    try:
+        fcntl.flock(fd, fcntl.LOCK_EX)
+    except Exception as exc:
+        raise CX(
+            f"Something unexpected happen while trying to acquire {lock_file}: {exc}"
+        )
+    else:
+        lock_file_fd = fd
+    finally:
+        if lock_file_fd is None:
+            os.close(fd)
+    yield lock_file_fd
+    if lock_file_fd:
+        fcntl.flock(lock_file_fd, fcntl.LOCK_UN)
+        os.close(lock_file_fd)
```

Comment:
I think this needs to go into a `finally` block to make sure it's executed when exceptions are raised

Reply:
Hmmm, yeah, that is correct. Thanks for the note!

I was initially under the impression that exceptions raised outside of the context manager (after yielding) were not preventing this code from being executed. I just double checked and you are correct, this is needed.